tests/unit/amo/components/TestAddonAuthorLinks.js

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+
+import AddonAuthorLinks, {
+  AddonAuthorLinksBase,
+} from 'amo/components/AddonAuthorLinks';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  dispatchClientMetadata,
+  dispatchSignInActions,
+  fakeAddon,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+  const slug = 'some-slug';
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  function render(props = {}) {
+    return shallowUntilTarget(
+      <AddonAuthorLinks
+        addon={props.addon || createInternalAddon(fakeAddon)}
+        i18n={fakeI18n()}
+        store={store}
+        {...props}
+      />,
+      AddonAuthorLinksBase,
+    );
+  }
+
+  const renderWithPermissions = ({
+    addon = createInternalAddon({
+      ...fakeAddon,
+      slug,
+      authors: [
+        {
+          id: 10642015,
+          name: 'avgupt',
+          picture_url: 'http://cdn.a.m.o/myphoto.jpg',
+          url: 'http://a.m.o/en-GB/firefox/user/avgupt/',
+          username: 'avgupt',
+        },
+      ],
+    }),
+    userId,
+  }) => {
+    dispatchSignInActions({ store, userId });
+    return render({ addon });
+  };
+
+  it('returns nothing if the userId is null / not logged in', () => {
+    const root = renderWithPermissions({});
+
+    expect(root.find('.AddonAuthorLinks')).toHaveLength(0);
+    expect(root.find('.AddonAuthorLinks-edit-link')).toHaveLength(0);
+  });
+
+  it('returns edit link if the author of the add-on in logged in', () => {
```

Comment:
Is this test case needed? We have a "shows an edit add-on link if the user is author" that looks great already.